bluepyopt/ephys/locations.py

```diff
@@ -392,6 +393,35 @@ def instantiate(self, sim=None, icell=None):
         return self.find_icomp(sim, branches)
 
 
+class NrnTrunkSomaDistanceCompLocation(NrnSecSomaDistanceCompLocation):
+    """Location at a distance from soma along a main direction.
+
+    This is most useful to follow the trunk of an apical dendrite
+    without knowing the apical point, but only that apical trunk goes along y.
+    """
+
+    def set_sec_index(self, icell=None, direction=[0.0, 1.0, 0.0]):
```

Comment:
Maybe I'm missing something, but isn't one supposed to specify the distance?

Reply:
ah this is just to find the sec index of the last section, then we use normal bpopt location which picks a section at a given location between soma and that section. Basically, instead of apical section id, we pick the furthest away, so we don't have to find the apical point, it's the accurate enough (if an apical trunk is <400, the dend2 feature we have will be in the tuft, but won't completely crash, or if apical point is spoted near soma as we have an early bifucration cell such as L5_TPC:B, then we'll pick one of the secondary apicla trunks, ...)

Reply:
Sorry, I confused set_sec_index with the constructor. So the original constructor is still there, all fine then

Reply:
But you're using euclidian distances, is that really what you want? Wouldn't one expect to use the path distance? (or was that just too difficult to implement?).
Actually, are you even using euclidian distances? Since you're allowing the user to specific any direction, shouldn't you have some norm equation in this?

Reply:
I just project on the vector direction, and take the largest value, I could add radial distance, or path distance, but I'm not sure it would be so useful. The direction is for example for horizontal or inverse PC cells.

Reply:
just to be sure, will this if one specifies a diagonal direction, or if the soma is not centered at 0,0,0?
I just want to make sure this is generally applicable if people specify a general direction.